pxr/usd/usdUtils/validators.cpp

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright 2024 Pixar
+//
+// Licensed under the terms set forth in the LICENSE.txt file available at
+// https://openusd.org/license.
+//
+
+#include <pxr/usd/sdf/fileFormat.h>
+#include "pxr/usd/usd/validationError.h"
+#include "pxr/usd/usd/validationRegistry.h"
+#include "pxr/usd/usdUtils/validatorTokens.h"
+#include "pxr/usd/ar/packageUtils.h"
+#include "pxr/usd/usdUtils/dependencies.h"
+#include "pxr/usd/usdUtils/userProcessingFunc.h"
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+    static
+    UsdValidationErrorVector
+    _PackageEncapsulationValidator(const UsdStagePtr& usdStage) {
+        UsdValidationErrorVector errors;
+
+        const SdfLayerRefPtr& rootLayer = usdStage->GetRootLayer();
+        const bool isPackage = [](const SdfLayerHandle& layer){
+            return layer->GetFileFormat()->IsPackage() || ArIsPackageRelativePath(layer->GetIdentifier());
+        }(rootLayer);
+
+        if (!isPackage){
+            return errors;
+        }
+
+        std::vector<TfRefPtr<SdfLayer>> layers;
+        std::vector<std::basic_string<char>> assets, unresolvedPaths;
+        const SdfAssetPath& path = SdfAssetPath(rootLayer->GetIdentifier());
+
+        UsdUtilsComputeAllDependencies(path, &layers, &assets,
+                                       &unresolvedPaths, nullptr);
+
+        const std::string& realPath = rootLayer->GetRealPath();
+        const std::string& packagePath = ArIsPackageRelativePath(rootLayer->GetIdentifier()) ?
+                                        ArSplitPackageRelativePathOuter(realPath).first :
+                                        realPath;
+
+        if (packagePath.length() > 0) {
+            for (const SdfLayerRefPtr& subLayer : layers) {
+                const std::string& realPath = subLayer->GetRealPath();
+                if (!TfStringStartsWith(realPath, packagePath)) {
+                    errors.emplace_back(
+                            UsdValidationErrorType::Error,
+                            UsdValidationErrorSites{
+                                    UsdValidationErrorSite(rootLayer,
+                                                           subLayer->GetDefaultPrimAsPath())
+                            },
+                            TfStringPrintf(("Found loaded layer '%s' that "
```

Comment:
I know the error message is coming from the current compliance check code. But why "loaded" layer? Maybe @dgovil or @spiffmon can clarify?

Reply:
I'm second guessing my decision to remove 'loaded' in the error message, also interested about it's meaning in this context.